### ISSUE TITLE
Segmentation fault on `prepare_cached` with an empty query

### ISSUE THREAD

**Issue by tomhoule** [jgallagher/rusqlite#583]
With the latest version, the following program segfaults on my machine:

```rust
use rusqlite::*;

fn main() -> Result<()> {
    let conn = Connection::open_in_memory()?;

    conn.prepare_cached("")?;

    Ok(())
}
```

rustqlite version: 0.20.0
sqlite version: 3.30.1-1
OS: Arch Linux

**Comment by gwenn** [comment]
Thanks you.
See PR #584 for a fix.
I am not sure how we should handle this case !

**Comment by tomhoule** [comment]
Should we close this issue, now that the PR is merged? :)

**Comment by gwenn** [comment]
PR is merged but not released yet.
I will try to release a new version this week.

**Comment by tomhoule** [comment]
No hurry from our side, we worked around the issue for now.

**Comment by gwenn** [comment]
Fixed in 0.21
